session/session.go

```diff
@@ -0,0 +1,83 @@
+// This package is for all 'session' related code.
+package session
+
+import (
+	"fmt"
+	"github.com/pocket_network/pocket-core/node"
+)
+
+var (
+	globalSessionList *sessionList
+)
+/*
+This is the session structure.
+ */
+ // TODO consider converting validators and servicers to map
+type Session struct {
+	// "devID" is the developer's ID that identifies the session.
+	devID string
+	// "validators" is an array of validator nodes.
+	validators []node.Validator
+	// "servicers" is an array of service nodes.
+	servicers []node.Service
+}
+
+/*
+This holds a list of sessions that are active (needs to confirm using liveness check).
+ */
+type sessionList struct {
+	// "sessions" is the local list of ongoing sessions.
+	sessions map[string]Session
+}
+
+/*
+ "GetSessionListInstance() returns the singleton instance of the global session list
+  TODO make thread safe
+ */
+func GetSessionListInstance() *sessionList {
+		if (globalSessionList == nil) {
+			globalSessionList = &sessionList{}
+			globalSessionList.sessions = make(map[string]Session)
+		}
+	return globalSessionList
+}
+
+/*
+"createNewSession" creates a new session for the specific devID and adds to global sessionList (map)
+ */
+func CreateNewSession(dID string) {
+	if(SearchSessionList(dID)==nil){
+		// pulls the global list from the singleton
+		sList :=GetSessionListInstance().sessions
```

Comment:
This isn't too important, but syntactically it looks awkward that you have to get the `sessions` property out of the `sessionList ` object, shouldn't the `sessionList` be the actual list of sessions?

Reply:
I purposefully did this so that the sessionList structure can become more robust in the future

Reply:
If you think that this is unnecessary then I can change it to just a list and not a custom object

Reply:
If that's the case, I think it would be best to create a class that extends list, and add the future functionality there. Keeping the list elements abstracted away and accessible to the list inherited functionality.

Reply:
Structures cannot be extended like classes as in other object-oriented programming languages. We can just create a new struct embedding the struct that we require to use from a different package.